examples/routing_service_shapes_processor/cpp/README.rst

```diff
@@ -19,25 +19,32 @@ From the directory containing the example sources:
 
     mkdir build
     cd build
-    cmake .. -DCONNEXTDDS_DIR=$NDDSHOME\
+    cmake .. -DCONNEXTDDS_DIR=<Connext directory> \
              -DCONNEXTDDS_ARCH=<ARCH> \
              -DBUILD_SHARED_LIBS=ON
-             -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
 
 
 where:
 
-- the environment variable ``$NDDSHOME`` shall point to the installation
-  directory of RTI Connext.
+- ``<Connext directory>`` shall be replaced with the path to the installation
+  directory of *RTI Connext*.
 - ``<ARCH>`` shall be replaced with the target architecture where you are
   running the example (e.g., x64Darwin15clang7.0).
 
+.. note::
+
+    Certain CMake generators may require specific value for the target platform
+    through the ``-DCMAKE_GENERATOR_PLATFORM``. For example on Windows system,
+    you can specify the values ``x86`` or ``x64``.
```

Comment:
I think you should add a link to [CMake Generators](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html).

Also, depending on the CMake version, the values for the generators are not the same. For instance:
``cmake -G "Visual Studio 15 2017 Win64"`` is now ``cmake -G "Visual Studio 15 2017" -A x64``

Note ``Win64`` was part of the generator name. Now, `` -A x64`` is a different option for CMake.